docs/docs/code/nodejs/using-data-stores/README.md

```diff
@@ -0,0 +1,169 @@
+---
+short_description: Store and read data with data stores.
+thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646763735/docs/icons/icons8-database-96_iv1oup.png
+---
+
+# Using Data Stores 
+
+In Node.js (Javascript) code steps, you can also store and retrieve data within code steps without connecting a 3rd party database. **Date Stores** are very useful for tracking data between runs of a particular workflow, making sure workflows only run once per unique record or sharing data between workflows.
+
+:::warning
+This functionality is limited to only Node.js code steps at this time.
+
+Other step languages like [Python](/code/python/), [Bash](/code/bash/) and [Go](/code/go/) do not have this feature available yet.
+
+For more information on what functionality is available for those languages, please refer to their documentation.
+:::
+
+### Adding a data store to a Node.js Step
+
+By default, Node.js steps don't have access to data stores. A data store can be added to your step by adding it as a `prop`.
+
+```javascript
+export default defineComponent({
+  props: {
+    // Define that the "db" variable in our component is a data store
+    data: { type: "store" }
+  },
+  async run({ steps, $ }) {
+    // Now we can access the data store at "this.store"
+    this.data.get("email");
+  }
+});
+```
+
+:::tip
```

Comment:
This is probably important for users to understand the code, so I would recommend moving it above the example and just integrate with the intro text. I added suggested language in the text above - lmk if that still captures what you want to communicate.